File: src/main/java/fi/metatavu/metaform/server/persistence/dao/TableReplyFieldNumberRowCellDAO.java
package fi.metatavu.metaform.server.persistence.dao;

import java.util.UUID;

import javax.enterprise.context.ApplicationScoped;

import fi.metatavu.metaform.server.persistence.model.TableReplyFieldNumberRowCell;
import fi.metatavu.metaform.server.persistence.model.TableReplyFieldRow;

/**
 * DAO class for TableReplyFieldNumberRowCell entity
 * 
 * @author Antti Leppä
 */
@ApplicationScoped
public class TableReplyFieldNumberRowCellDAO extends AbstractTableReplyFieldRowCellDAO<TableReplyFieldNumberRowCell> {

  /**
   * Creates new tableReplyFieldNumberRowCell
   *
   * @param id id
   * @param row row
   * @param value value
   * @param name name
   * @return created tableReplyFieldNumberRowCell
   */
  public TableReplyFieldNumberRowCell create(UUID id, TableReplyFieldRow row, String name, Double value) {
    TableReplyFieldNumberRowCell tableReplyFieldNumberRowCell = new TableReplyFieldNumberRowCell();
    tableReplyFieldNumberRowCell.setId(id);
    tableReplyFieldNumberRowCell.setRow(row);
    tableReplyFieldNumberRowCell.setName(name);
    tableReplyFieldNumberRowCell.setValue(value);
    return persist(tableReplyFieldNumberRowCell);
  }

  /**
  * Updates value
  *
  * @param value value
  * @return updated tableReplyFieldNumberRowCell
  */
  public TableReplyFieldNumberRowCell updateValue(TableReplyFieldNumberRowCell tableReplyFieldNumberRowCell, Double value) {
    tableReplyFieldNumberRowCell.setValue(value);
    return persist(tableReplyFieldNumberRowCell);
  }

}

File: src/main/kotlin/fi/metatavu/metaform/persistence/dao/TableReplyFieldNumberRowCellDAO.kt
package fi.metatavu.metaform.persistence.dao

import fi.metatavu.metaform.persistence.model.TableReplyFieldNumberRowCell
import fi.metatavu.metaform.persistence.model.TableReplyFieldRow
import java.util.*
import javax.enterprise.context.ApplicationScoped

/**
 * DAO class for TableReplyFieldNumberRowCell entity
 *
 * @author Antti Leppä
 */
@ApplicationScoped
class TableReplyFieldNumberRowCellDAO :
  AbstractTableReplyFieldRowCellDAO<TableReplyFieldNumberRowCell>() {
  /**
   * Creates new tableReplyFieldNumberRowCell
   *
   * @param id id
   * @param row row
   * @param value value
   * @param name name
   * @return created tableReplyFieldNumberRowCell
   */
  fun create(
    id: UUID,
    row: TableReplyFieldRow,
    name: String,
    value: Double
  ): TableReplyFieldNumberRowCell? {
    val tableReplyFieldNumberRowCell = TableReplyFieldNumberRowCell()
    tableReplyFieldNumberRowCell.id = id
    tableReplyFieldNumberRowCell.row = row
    tableReplyFieldNumberRowCell.name = name
    tableReplyFieldNumberRowCell.value = value
    return persist(tableReplyFieldNumberRowCell)
  }

  /**
   * Updates value
   *
   * @param value value
   * @return updated tableReplyFieldNumberRowCell
   */
  fun updateValue(
    tableReplyFieldNumberRowCell: TableReplyFieldNumberRowCell,
    value: Double
  ): TableReplyFieldNumberRowCell? {
    tableReplyFieldNumberRowCell.value = value
    return persist(tableReplyFieldNumberRowCell)
  }
}